docs/src/guides/advanced/18_allowlist.md

```diff
@@ -0,0 +1,114 @@
+# Contract Deployment Allowlist
+
+This module restricts contract deployments on zkSync Era to a predefined list of approved addresses.
+
+It consists of:
+
+- `SharedAllowList`: in-memory, async-safe address set.
+- `AllowListTask`: periodically fetches the list from a remote URL.
+- `WhitelistedDeployPoolSink`: enforces the list during L2 transaction submission.
+
+---
+
+## Configuration
+
+Configure via `api.web3_json_rpc.deployment_allowlist` in your node's YAML:
+
+```api:
+  web3_json_rpc:
+    deployment_allowlist:
+      http_file_url: "https://example.com/allowlist.json"
+      refresh_interval_secs: 60
+```
+
+- `http_file_url` (required): HTTP(S) URL to the allowlist JSON.
+- `refresh_interval_secs` (optional): Refresh interval in seconds (default: 60).
+
+To disable, set:
+
+```
+api:
+  web3_json_rpc:
+    deployment_allowlist: {}
+```
+
+---
+
+## Allowlist Format
+
+The JSON file must contain:
+
+```
+{
+  "addresses": [
+    "0x1234...abcd",
+    "0xbeef...dead"
+  ]
+}
+```
+
+- Addresses must be checksummed.
```

Comment:
What does this mean? AFAICT, no checksum verification is performed on the app level.